### DISCUSSION TITLE
Can I config to make this constantly sync my OneDrive files to Ubuntu?

### DISCUSSION THREAD

**Discussion by GalaxiumNation** [abraunegg/onedrive#2685] [Q&A] [answered]
How do I config so onedrive constantly checks and syncs files when I upload a new file to _Microsoft_ OneDrive? If so how should/do I configure this?
And will this sync any new file I put in "OneDrive" folder (on ubuntu side) to OneDrive automatically? Because I'm trying to sync my memes from ubuntu to onedrive without having to upload them to onedrive in my browser.

**Comment by abraunegg** [comment, marked as answer]
@GalaxiumNation 
> How do I config so onedrive constantly checks and syncs files when I upload a new file to Microsoft OneDrive? If so how should/do I configure this?

This is clearly documented in the documentation, as you must run the client as a systemd service. Please read:
* https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#running-onedrive-as-a-system-service
* https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-release-candidate-1/docs/usage.md#how-to-initiate-the-client-as-a-background-service

> And will this sync any new file I put in "OneDrive" folder (on ubuntu side) to OneDrive automatically?

When a systemd service is setup, the local filesystem is monitored for changes using 'inotify'. Once a change has been detected, this local change is uploaded to Microsoft OneDrive.

For further details on how the client works, please read: https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-release-candidate-1/docs/client-architecture.md

**Reply by GalaxiumNation** [reply]
Ok so the files i put in onedrive folder on ubuntu do sync over to the official, but I'm still having trouble with it syncing constantly.
I wan it to sync at least every 30 minutes.

**Comment by abraunegg** [comment]
@GalaxiumNation 
OK .. lets start with the basics:
1. How did you install the client? Given you were following that 'gist' initially .. have you *actually* installed the client correctly? 
2. What is your client version that you are running `onedrive --version`
3. Check your actual system if all the systemd service files have been installed: `systemctl list-units --type=service --all | grep onedrive`
